src/test/kotlin/net/revanced/patcher/PatcherTest.kt

```diff
@@ -110,39 +114,25 @@ internal class PatcherTest {
         )
 
         // Apply all patches loaded in the patcher
-        val result = patcher.applyPatches()
+        val patchResult = patcher.applyPatches()
         // You can check if an error occurred
-        for ((s, r) in result) {
-            if (r.isFailure) {
-                throw Exception("Patch $s failed", r.exceptionOrNull()!!)
+        for ((patchName, result) in patchResult) {
+            if (result.isFailure) {
+                throw Exception("Patch $patchName failed", result.exceptionOrNull()!!)
             }
         }
 
-        // TODO Doesn't work, needs to be fixed.
-        //val out = ByteArrayOutputStream()
-        //patcher.saveTo(out)
-        //assertTrue(
-        //    // 8 is a random value, it's just weird if it's any lower than that
-        //    out.size() > 8,
-        //    "Output must be at least 8 bytes"
-        //)
-        //
-        //out.close()
-        testData.close()
+        patcher.save()
     }
 
-    // TODO Doesn't work, needs to be fixed.
-    //@Test
-    //fun `test patcher with no changes`() {
-    //    val testData = PatcherTest::class.java.getResourceAsStream("/test1.jar")!!
-    //    val available = testData.available()
-    //    val patcher = Patcher(testData, testSigs)
-    //
-    //    val out = ByteArrayOutputStream()
-    //    patcher.saveTo(out)
-    //    assertEquals(available, out.size())
-    //
-    //    out.close()
-    //    testData.close()
-    //}
+    @Test
+    fun `test patcher with no changes`() {
+        val patcher = Patcher(
+            PatcherTest::class.java.getResourceAsStream("/test1.jar")!!,
+            ByteArrayOutputStream(),
```

Comment:
Make the output stream a variable and test that with no changes the output should be the same as the input. (you removed this code, see the PR diff to check how I did it previously or do it your own way)

Reply:
I an not sure if they are gonna be the same, since we rewrite into a new .jar file, so this might not make much sense

Reply:
That's true. I have an idea though, I'll checkout this PR locally and test if my idea works. If so I'll commit it to the PR.